.github/workflows/release-please.yaml

```diff
@@ -0,0 +1,88 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    outputs:
+      release_created: ${{ steps.release.outputs.release_created }}
+      tag_name: ${{ steps.release.outputs.tag_name }}
+      patch: ${{ steps.release.outputs.patch }}
+
+    steps:
+      # Create/update release PR
+      - uses: googleapis/release-please-action@v4
+        id: release
```

Comment:
does this mean we will have a release PR with every push to `main`? should it be dispatch only instead?

Reply:
Yes and no.  There will be a release PR after every conventional commits change that matters (fix & feat only IIRC).  We don't have to merge it right away, though.  If we merge additional changes, the release PR gets updated (version number, changelog).

The only things in the release PR are changelog and version number updates.  The version number update is based on the commits (fix only, bump patch number; feat, bump minor number; feat! (breaking change), bump major number).

We only merge the release PR when we are ready to release.  After merging the release PR, the actual release steps will execute: create GitHub release, attach artifacts to release, publish to PyPi.